src/wp-includes/bookmark-template.php

```diff
@@ -102,13 +102,27 @@ function _walk_bookmarks( $bookmarks, $args = '' ) {
 			$title = ' title="' . $title . '"';
 		}
 		$rel = $bookmark->link_rel;
-		if ( '' !== $rel ) {
-			$rel = ' rel="' . esc_attr( $rel ) . '"';
-		}
+
 		$target = $bookmark->link_target;
 		if ( '' !== $target ) {
+			if ( is_string( $rel ) && '' !== $rel ) {
+				$rels   = explode( ' ', $rel );
+
+				if ( in_array( 'noopener', $rels, true ) ) {
+					$rels[] = 'noopener';
+					$rel     = implode( ' ', $rels );
+				}
```

Comment:
Replace with less expensive processing:

```suggestion
				if ( ! str_contains( $rels, 'noopener' ) ) {
					$rel = trim( $rel ) . ' noopener';
				}
```

A polyfill was added today to Core for using `str_contains()` (i.e. a new PHP 8.0 function). This function eliminates parsing into an array, searching the array, and then converting the array back into a string. Less memory and processing.